openfeature/client.go

```diff
@@ -713,16 +713,18 @@ func (c *Client) evaluate(
 		c.finallyHooks(ctx, hookCtx, providerInvocationClientApiHooks, options)
 	}()
 
-	// short circuit if provider is in NOT READY state
-	if c.State() == NotReadyState {
-		c.errorHooks(ctx, hookCtx, providerInvocationClientApiHooks, ProviderNotReadyError, options)
-		return evalDetails, ProviderNotReadyError
-	}
+	if _, ok := provider.(NoopProvider); !ok {
```

Comment:
Bypass short-circuit logic for the Noop provider, as it is essentially stateless.